### PR TITLE
feat: batching in engine-v2 in cli

### PR DESCRIPTION
**Pull request by obmarg**
As the title says.

Fixes GB-6026

### CODE REVIEW

File: cli/crates/federated-dev/src/dev.rs [hunk 1]
```diff
@@ -200,37 +203,61 @@ async fn handle_engine_request(
         headers,
         wait_until_sender: sender,
     };
-    request.ray_id = ctx.ray_id.clone();
+    let ray_id = ctx.ray_id.clone();
+    // request.ray_id = ctx.ray_id.clone();
 
-    let session = gateway.authorize(ctx.headers_as_map().into()).await;
+    if matches!(request, BatchRequest::Batch(_)) && streaming_format.is_some() {
+        todo!("Error")
```
Comment (obmarg) [inline cli/crates/federated-dev/src/dev.rs:210]: Need to do this

File: cli/crates/federated-dev/src/dev.rs [hunk 2]
```diff
@@ -200,37 +203,61 @@ async fn handle_engine_request(
...
-                    streaming_format,
+                    format,
                 )
-                .await
+                .await;
+
+                return (headers, axum::body::Body::from_stream(stream)).into_response();
+            }
+            (BatchRequest::Batch(_), _) => {
+                todo!("return multiple errors")
```
Comment (obmarg) [inline cli/crates/federated-dev/src/dev.rs:231]: Need to do this

File: cli/crates/federated-dev/src/dev/batch_response.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            BatchResponse::Single(response) => (response.status, response.headers, response.bytes).into_response(),
+            BatchResponse::Batch(responses) => {
+                // These are awkward af to deal with - they have bytes inside and we need to make
+                // them into a JSON list. So we're going to have to do a serde roundtrip.
+                // Inefficient, but I don't have time to go refactoring everything rn. It's also
+                // unclear that avoiding this would actually be more efficient overall, because the
+                // response containing bytes is itself an optimisation - avoids serializing twice
+                // when caching and avoids a serde roundtrip when reading from the cache.  Urgh,
+                let Ok((headers, bytes)) = headers_and_body(responses, serialized_size) else {
+                    todo!("Error")
```
Comment (obmarg) [inline cli/crates/federated-dev/src/dev/batch_response.rs:24]: Need to do this

**Review by tomhoule — COMMENTED** [review]
:100:

**Review by tomhoule — APPROVED** [review]

**Pull request merged**